.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+default_language_version:
+  python: python3.8
+
+ci:
+  autofix_prs: true
+  autoupdate_commit_msg: '[pre-commit.ci] pre-commit suggestions'
+  # submodules: true
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-yaml
+      - id: pretty-format-json
+
+  - repo: https://github.com/pre-commit/mirrors-yapf
+    rev: v0.31.0
+    hooks:
+      - id: yapf
+        name: formatting
+        entry: yapf
+        args: [--parallel]
```

Comment:
by the way, you're fork bombing -- pre-commit runs things in parallel by default (and tends to do a better job than `yapf` does)